pkg/apis/core/annotation_key_constants.go

```diff
@@ -83,6 +83,13 @@ const (
 	// Not all cloud providers support this annotation, though AWS & GCE do.
 	AnnotationLoadBalancerSourceRangesKey = "service.beta.kubernetes.io/load-balancer-source-ranges"
 
+	// AnnotationLoadBalancerAllowMixedProtocolsKey is the key of the annotation on a service to allow using mixed protocols (e.g. TCP+UDP) on the same LoadBalancer
+	//
+	// It should be a string with either "true", or "false".
+	//
+	// Not all cloud providers support this annotation, though Azure, OpenStack and MetalLB do.
+	AnnotationLoadBalancerAllowMixedProtocolsKey = "service.beta.kubernetes.io/load-balancer-allow-mixed-protocols"
```

Comment:
Can you expand a little on the reasoning for an annotation? A check if a Service contains ports for more than one protocol shouldn't be too hard to write and could be used as a basis for having providers that do not support that emit an event. Or are there more reasons for having it?

Reply:
Really, I think the annotation should be little more than a stopgap measure to handle compatibility and allow people (i.e. CPI vendors) time to update - I gave some detail in the PR message above.